File: src/summary/summary.model.go
package summary

import (
	"github.com/ZombieHippie/test-gin/src/artifact"
	"github.com/ZombieHippie/test-gin/src/repo"
	"github.com/jinzhu/gorm"
	"time"
)

// Summary is a common object for reports to point at
type Summary struct {
	gorm.Model
	Repository    repo.Repository
	BranchID      string
	PullRequestID int
	BuildID       int
	Artifacts     []artifact.Artifact
	Commit        string

	Message string
	Author  string
	Success bool
	Created time.Time
}

File: src/summary/summary.model.ts

import { Repository } from "../repo/repo.model"
import { Artifact } from "../artifact/artifact.model"

interface Summary {
  ID?: number // Assigned by gorm
  Repository: Repository // Assign this yourself
  BuildID: number // 2
  PullRequestID: number // 5
  BranchID: string // "feature/no-more-bugs"
  Artifacts: Artifact[] // Files you want to attach
  Commit: string // Commit hash
  Success: boolean
  Created: Date // Date Summary was recorded
  Message?: string
  Author?: string
}

export { Summary }

